controller_interface/src/chainable_controller_interface.cpp

```diff
@@ -87,11 +88,24 @@ ChainableControllerInterface::export_state_interfaces()
       throw std::runtime_error(error_msg);
     }
     ordered_exported_state_interfaces_.push_back(state_interface);
-    exported_state_interface_names_.push_back(interface_name);
+    add_element_to_list(exported_state_interface_names_, interface_name);
     state_interfaces_ptrs_vec.push_back(
       std::const_pointer_cast<const hardware_interface::StateInterface>(state_interface));
   }
 
+  if (exported_state_interfaces_.size() != state_interfaces.size())
+  {
+    std::string error_msg =
+      "The internal storage for state interface ptrs 'exported_state_interfaces_' variable has "
+      "size '" +
+      std::to_string(exported_state_interfaces_.size()) +
+      "', but it is expected to have the size '" + std::to_string(state_interfaces.size()) +
+      "' equal to the number of exported reference interfaces. Please correct and recompile the "
+      "controller with name '" +
+      get_node()->get_name() + "' and try again.";
+    throw std::runtime_error(error_msg);
+  }
+
```

Comment:
could you please elaborate on when this may happen?

Reply:
Probably, it won't. I just tried to maintain the same checks as in the other export method

https://github.com/ros-controls/ros2_control/blob/a6e44abcfdc9a1246ac1c3e0d0e0091226f67953/controller_interface/src/chainable_controller_interface.cpp#L163-L173

If we don't need it, I can remove it from the other export method too

Reply:
ok